### PR TITLE
GETKEYSANDFLAGS sub-command, new module apis, and key-spec flags changes

### PR DESCRIPTION
**Pull request by oranagra**
doc changes for https://github.com/redis/redis/pull/10237

### CODE REVIEW

File: commands.json [hunk 1]
```diff
@@ -2854,6 +2855,21 @@
             "stale"
         ]
     },
+    "COMMAND GETKEYSANDFLAGS": {
+        "summary": "Extract keys given a full Redis command",
```
Comment (itamarhaber) [inline commands.json:2859]:
```suggestion
        "summary": "Extract keys and flags given a full Redis command",
```

File: commands.json [hunk 2]
```diff
@@ -4244,7 +4260,7 @@
         "arity": 3,
         "arguments": [
             {
-                "name": "function-name",
+                "name": "library-name",
```
Comment (itamarhaber) [inline commands.json:4263]: 👍

**Review by itamarhaber — APPROVED** [review]

File: topics/key-specs.md [hunk 3]
```diff
@@ -180,6 +181,12 @@ The difficulty arises, for example, because the string _"STORE"_ is both a keywo
 the only commands with _incomplete_ key specifications are `SORT` and `MIGRATE`.
 We don't expect the addition of such commands in the future.
 
+### variable_flags
+
+In some commands the flags for the same key name argument can depend on other arguments.
+For example the SET command has a GET argument, without the GET argument, SET is a write-only command, but with it, it becomes read+write.
+When this flag is present, it means that the flags in the key-spec will cover all options, but depending on some arguments, some flags can be removed or changed.
```
Comment (itamarhaber) [inline topics/key-specs.md:188]:
```suggestion
In some commands, the flags for the same key name argument can depend on other arguments.
For example, consider the `SET` command and its optional  _GET_ argument.
Without the _GET_ argument, `SET` is write-only, but it becomes a read and write command with it.
When this flag is present, it means that the key specification flags cover all possible options, but the effective flags depend on other arguments.
```

File: commands.json (hunk 1 above)
Comment (oranagra) [reply commands.json:2859]: @itamarhaber should i remind you this is an auto-generated file?
i can apply your suggestion on the source, but i'm too lazy, and i think it's clear without it..

File: commands.json (hunk 2 above)
Comment (itamarhaber) [reply commands.json:4263]: :)

**Pull request merged**